src/sme_lib/enemy/FireyPetey.cpp

```diff
@@ -87,12 +84,10 @@ void TFPTornado::perform(u32 flags, JDrama::TGraphics *graphics)
     return;
   }
   TBPTornado::perform(flags, graphics);
-  *(float *)((char *)this + 0x70) = gpMarioPos->x;
-  *(float *)((char *)this + 0x74) = gpMarioPos->y;
-  *(float *)((char *)this + 0x78) = gpMarioPos->z;
-  for (int iVar9 = 0; iVar9 < mNumObjs; iVar9 = iVar9 + 1)
+  mTargetPos = *gpMarioPos;
+  for (int i = 0; i < mNumObjs; i = i + 1)
   {
-    if (mCollidingObjs[iVar9]->mObjectID == -0x7fffffff)
+    if (mCollidingObjs[i]->mObjectID == -0x7fffffff)
```

Comment:
Use the define here instead of a magic number. Include from `sms/actor/Mario.hxx` and use `OBJECT_ID_MARIO`

You will likely need to pull this in from main.